docs/imageMgmtAPI.rst

```diff
@@ -0,0 +1,373 @@
+|  imageType      |   String    |  Image Type Name (PK)                |
++-----------------+-------------+--------------------------------------+
+|  description    |   String    |  Description of the image type       |
++-----------------+-------------+--------------------------------------+
+|  deployable     |   enum      |  IMAGE (docker image), JAR or TAR    |
++-----------------+-------------+--------------------------------------+
+|  ownerships     | JSON String | List of:                             |
+|                 |             | {"owner": <user-id>, "role": <role>} |
+|                 |             |                                      |
+|                 |             | Possible roles: ``ADMIN``,           |
+|                 |             | ``MEMBER`` or ``GUEST``              |
++-----------------+-------------+--------------------------------------+
+
+**Response:**
+
+.. code-block::
+
+  Status: 201 Created
+  Header -> Location: /imageTypes/{id}
+
+/imageVersions
+**************
+This endpoint is used to register or list image versions for a given image type. The API
+assumes that image type is already registered (Create API for /imageTypes was called before) with Azkaban.
+
+The image binary must already be uploaded to a location in artifactory, accessible to Azkaban. Azkaban
+does not explicitly check if the image is in fact available. It is the image owner's responsibility to
+ensure it's availability prior to calling this API. By calling this API, the image
+admin is merely informing Azkaban for the availability of a given image version.
+
+.. _create-image-version:
+
+Create (Register) new Image Version
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+- **Method:** POST
+- **Request URL:** /imageVersions
+- **Request Body:**
+
++-----------------+-------------+-------------------------------------------------+
+|   Field Name    |     Type    |            Description                          |
++=================+=============+=================================================+
+|  imagePath      |   String    |                                                 |
++-----------------+-------------+-------------------------------------------------+
+|  imageVersion   |   String    |                                                 |
++-----------------+-------------+-------------------------------------------------+
+|  description    |   String    |                                                 |
++-----------------+-------------+-------------------------------------------------+
+|  versionState   |   String    | ``NEW``, ``ACTIVE``, ``UNSTABLE``,              |
+|                 |             | ``DEPRECATED``                                  |
++-----------------+-------------+-------------------------------------------------+
+|  releaseTag     |   String    |                                                 |
++-----------------+-------------+-------------------------------------------------+
```

Comment:
Example payload for create version:

{
    "imagePath": "full image path in the artifactory",
    "imageVersion": "1.6.1",
    "imageType": "spark",
    "description": "spark new version",
    "versionState": "NEW",
    "releaseTag": "1.5.9"
}